feeds/wlan-ap/opensync/src/platform/openwrt/src/netifd/src/wifi_inet_config.c

```diff
@@ -284,15 +293,23 @@ static int wifi_inet_conf_add(struct schema_Wifi_Inet_Config *iconf)
 	}
 
 	if (!strcmp(iconf->if_type, "gre")) {
-		blobmsg_add_string(&b, "ifname", iconf->gre_ifname);
 		if (!is_ipv6) {
+			char ip_addr[IPV4_ADDR_STR_LEN];
+			if (ubus_get_wan_ip(ip_addr) || !strlen(ip_addr)) {
+				LOG(ERR, "netifd: Failed to get wan ip. GRE tunnel interface won't be created");
+				return -1;
+			}
 			blobmsg_add_string(&b, "proto", "gretap");
-			blobmsg_add_string(&b, "ipaddr", iconf->gre_local_inet_addr);
+			blobmsg_add_string(&b, "ipaddr", ip_addr);
 			blobmsg_add_string(&b, "peeraddr", iconf->gre_remote_inet_addr);
+			blobmsg_add_u32(&b, "mtu", 1500);
+			blobmsg_add_bool(&b, "force_link", 1);
 		} else {
 			blobmsg_add_string(&b, "proto", "grev6tap");
 			blobmsg_add_string(&b, "ip6addr", iconf->gre_local_inet_addr);
```

Comment:
Don't understand why behaviour is different for Iv4 and IPv6 addresses.

Reply:
Again, v6 is not fully supported yet.